internal/services/azurestackhci/stack_hci_logical_network_resource_test.go

```diff
@@ -0,0 +1,333 @@
+package azurestackhci_test
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/azurestackhci/2024-01-01/logicalnetworks"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance/check"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type StackHCILogicalNetworkResource struct{}
+
+// https://learn.microsoft.com/en-us/azure-stack/hci/manage/create-logical-networks?tabs=azurecli#prerequisites
+// The resource can only be created on the customlocation generated after HCI deployment
+const (
+	customLocationIdEnv = "ARM_TEST_STACK_HCI_CUSTOM_LOCATION_ID"
+)
+
+func TestAccStackHCILogicalNetwork_dynamic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_stack_hci_logical_network", "test")
+	r := StackHCILogicalNetworkResource{}
+
+	if os.Getenv(customLocationIdEnv) == "" {
+		t.Skipf("skip the test as one or more of below environment variables are not specified: %q", customLocationIdEnv)
```

Comment:
I only see one required environment variable for the test: 
```suggestion
		t.Skipf("skipping since %q has not been specified", customLocationIdEnv)
```

Reply:
changed